modules/stream/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/utils.scala

```diff
@@ -78,23 +86,54 @@ object utils {
-                      "accessKey and secretKey must both be set to 'iam' or neither".asLeft
-                    case AWSCredentials(a, s) if isEnv(a) && isEnv(s) =>
-                      new EnvironmentVariableCredentialsProvider().asRight
-                    case AWSCredentials(a, s) if isEnv(a) || isEnv(s) =>
-                      "accessKey and secretKey must both be set to 'env' or neither".asLeft
-                    case AWSCredentials(a, s) =>
-                      new AWSStaticCredentialsProvider(new BasicAWSCredentials(a, s)).asRight
-                  }
+        case NoCredentials => "No AWS credentials provided".asLeft
+        case _: GCPCredentials => "GCP credentials provided".asLeft
+        case AWSCredentials(a, s) if isDefault(a) && isDefault(s) =>
+          new DefaultAWSCredentialsProviderChain().asRight
+        case AWSCredentials(a, s) if isDefault(a) || isDefault(s) =>
+          "accessKey and secretKey must both be set to 'default' or neither".asLeft
+        case AWSCredentials(a, s) if isIam(a) && isIam(s) =>
+          AWSInstanceProfileCredentialsProvider.getInstance().asRight
+        case AWSCredentials(a, s) if isIam(a) && isIam(s) =>
+          "accessKey and secretKey must both be set to 'iam' or neither".asLeft
+        case AWSCredentials(a, s) if isEnv(a) && isEnv(s) =>
+          new AWSEnvironmentVariableCredentialsProvider().asRight
+        case AWSCredentials(a, s) if isEnv(a) || isEnv(s) =>
+          "accessKey and secretKey must both be set to 'env' or neither".asLeft
+        case AWSCredentials(a, s) =>
+          new AWSStaticCredentialsProvider(new BasicAWSCredentials(a, s)).asRight
+      }
+    } yield provider
+  }
+
+  def getAwsCredentialsProvider(creds: Credentials): Either[String, AwsCredentialsProvider] = {
+    def isDefault(key: String): Boolean = key == "default"
+    def isIam(key: String): Boolean = key == "iam"
+    def isEnv(key: String): Boolean = key == "env"
+
+    for {
+      provider <- creds match {
+        case NoCredentials => "No AWS credentials provided".asLeft
+        case _: GCPCredentials => "GCP credentials provided".asLeft
+        case AWSCredentials(a, s) if isDefault(a) && isDefault(s) =>
+          DefaultCredentialsProvider.create().asRight
+        case AWSCredentials(a, s) if isDefault(a) || isDefault(s) =>
+          "accessKey and secretKey must both be set to 'default' or neither".asLeft
+        case AWSCredentials(a, s) if isIam(a) && isIam(s) =>
+          InstanceProfileCredentialsProvider.create().asRight
+        case AWSCredentials(a, s) if isIam(a) && isIam(s) =>
+          "accessKey and secretKey must both be set to 'iam' or neither".asLeft
+        case AWSCredentials(a, s) if isEnv(a) && isEnv(s) =>
+          EnvironmentVariableCredentialsProvider.create().asRight
+        case AWSCredentials(a, s) if isEnv(a) || isEnv(s) =>
+          "accessKey and secretKey must both be set to 'env' or neither".asLeft
```

Comment:
Duplicate code exists in `getAWSCredentialsProvider` and this method.

If we are going to migrate both consumer and producer, this comment isn't valid. Otherwise, we need to deduplicate this snippet.